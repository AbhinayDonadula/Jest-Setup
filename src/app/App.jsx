import PropTypes from 'prop-types';
import React from 'react';

function App({ title }) {
    return (
        <div>
            <span>{title}</span>
        </div>
    );
}
App.propTypes = {
    title: PropTypes.string,
};

App.defaultProps = {
    title: 'Abhinay',
};
export default App;
