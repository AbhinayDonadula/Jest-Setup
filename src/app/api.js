export const getUser = () => {
    console.log('getting user');

    return fetch('/user').then((resp) => resp.json());
};
