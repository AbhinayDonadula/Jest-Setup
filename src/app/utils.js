export const add = (a, b) => a + b;
export const sub = (a, b) => a - b;

export const getAbove3GPAStuds = (students = []) => {
    const a = 20;
    const result = students.filter((each) => each.gpa > 3);
    return result;
};
