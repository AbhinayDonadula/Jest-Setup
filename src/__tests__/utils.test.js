import { getAbove3GPAStuds } from '../app/utils';

const students = [
    { name: 'st1', gpa: 3 },
    { name: 'st2', gpa: 2 },
    { name: 'st3', gpa: 1 },
    { name: 'st4', gpa: 4 },
    { name: 'st7', gpa: 3.1 },
    { name: 'st5', gpa: 2 },
    { name: 'st6', gpa: 1 },
];

const testHTML = "<div class='a'>Testing</div>";

test('get 3 or above gpa students', () => {
    const a = 30;
    expect(getAbove3GPAStuds(students)).toBeArrayOfSize(2);

    expect(getAbove3GPAStuds([])).toBeArrayOfSize(0);

    expect(getAbove3GPAStuds()).toBeArrayOfSize(0);

    expect(getAbove3GPAStuds(students)).toBeArray();

    expect(getAbove3GPAStuds(students)).toMatchSnapshot();

    expect(getAbove3GPAStuds(students)).toMatchInlineSnapshot(`
        Array [
          Object {
            "gpa": 4,
            "name": "st4",
          },
          Object {
            "gpa": 3.1,
            "name": "st7",
          },
        ]
    `);

    expect(testHTML).toMatchSnapshot();
});
