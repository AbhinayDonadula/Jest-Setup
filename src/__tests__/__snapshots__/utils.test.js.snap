// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get 3 or above gpa students 1`] = `
Array [
  Object {
    "gpa": 4,
    "name": "st4",
  },
  Object {
    "gpa": 3.1,
    "name": "st7",
  },
]
`;

exports[`get 3 or above gpa students 3`] = `"<div class='a'>Testing</div>"`;
